Sources/SwifterSwift/SwiftStdlib/DictionaryExtensions.swift

```diff
@@ -189,6 +189,28 @@ public extension Dictionary {
             }
         }
     }
+    
+    /// SwifterSwift: Filters out a new dictionary based on spericied `keys`
+    ///
+    ///        var dict =  ["key1": 1, "key2": 2, "key3": 3, "key4": 4]
+    ///        dict["key1", "key3", "key4"] -> ["key1": 1, "key3": 3, "key4": 4]
+    ///        dict["key2"] -> ["key2": 2]
+    ///
+    /// - Note: A new copy of a dictionary is created. if none of the keys are found an empty dictionary will be returned
+    ///
+    /// - Complexity: O(K), _K  is the length of the keys array and O(K) space.
+    ///
+    /// - Parameter keys: An array of keys to the desired value.
+    ///
+    /// - Returns: A new dictionary that contains specified  keys only . if none of the keys exists an empty dictioanry will be returned.
+    subscript(withKeys keys: Key...) -> [Key: Value] {
```

Comment:
Not sure having this as a subscript is an ideal API... This may probably be a better suit to go for a `filter(byKeys: Key...)`.
But at the same time, I'm not sure if such a version of `filter` is worth adding since we can achieve the same with very simple code 
```swift 
let dict = ["a": 1, "b": 2]
let keys = ["a"]
let filtered = dict.filter { keys.contains($0.key) }
print(filtered) // ["a": 1]
```
So this would feel like we are abstracting something that is very simple already...

Reply:
Hi @LucianoPAlmeida  
The main idea is to filter out a certain "slice" of a Dictionary based on a given array of keys only, `filter` is a good candidate, but in this case, it might not be an efficient solution, as we need to iterate over all of the `(key, value)` tuples, which takes `O(N)`, then for each iteration, we need to look up for the current `$0.key` in `keys` which takes `O(K)` where K is the number of keys, what gives in a total running time of `O(N*K)`.

The name and subscript can be changed to a function in case we will proceed with this PR :)

Reply:
I see that this makes sense, but still not sure about it being a subscript. IMO it should be a function...

Reply:
This could be useful. I agree with Luciano - subscript is not a good fit. When in doubt, I look to lodash. Perhaps a function called `pick`? See https://lodash.com/docs/4.17.15#pick.

Reply:
@gurgeous 
Nice idea, I'll change it to be a function and name it as `pick`

Reply:
@MaratIbragimov to fix the `O(N*K)` issue you just need to create a `Set` so that it becomes `O(N+K)`.
```Swift
let keys = Set(["a"])
```

Reply:
@guykogus indeed, that's what I did at the beginning, but then I changed it, because I actually have an O(1) access on the dictionary without additional allocation.
The only case I can think of using `Set` is when there are duplicated keys passed to the` function` / `subscript`

Reply:
@MaratIbragimov you're right. If you just traverse through the keys you can check the dictionary directly.
```Swift
func pick(keys: [Key]) -> Dictionary {
    return keys.reduce(into: Dictionary()) { result, key in
        result[key] = self[key]
    }
}
```
Otherwise you have to first go through all the keys and then through the entire dictionary.
```Swift
func pick(keys: [Key]) -> Dictionary {
    let keysSet = Set(keys)
    return filter { key, _ in keysSet.contains(key) }
}
```
If you want to get really fancy, you can make this a one-liner. It's a bit ugly though 😬
```Swift
func pick(keys: [Key]) -> Dictionary {
    return keys.reduce(into: Dictionary()) { $0[$1] = self[$1] }
}
```

Reply:
Thanks :), @guykogus I guess I'll go with your first suggestion, as it reads better in my opinion :)